src/block_helpers.rs

```diff
@@ -1,19 +1,19 @@
-#![allow(dead_code,unused_imports)]
-
 use crate::optimizer::OptimizerTrait;
 use std::error::Error;
 use std::io;
 use std::io::Read;
 
 use crate::feature_buffer;
-use crate::graph;
 use crate::optimizer::OptimizerSGD;
 use crate::port_buffer;
 use crate::regressor::{BlockCache, BlockTrait};
 use std::cmp::min;
 use std::mem;
 use std::slice;
 
+#[cfg(test)]
+use crate::graph;
```

Comment:
Not sure I understand this one - is `graph` used only for tests?

Reply:
Yes, the `graph` is only used in the methods that are made & used in tests (slearn2, ssetup_cache2, ....). & this way it only imports it with when tests are run.